### ISSUE TITLE
Can't seem to use other components as items

### ISSUE THREAD

**Issue by mattbishop** [Jexordexan/vue-slicksort#72]
I want to create a sortable list of component elements. I can get them to render fine, but after I drag, the elements repopulate with their original sorted value. It appears the v-model binding is not being updated down or something.

**Comment by mattbishop** [comment]
For a little more detail, if I change the state of the list component, it will update the items list in SlickList as expected. However, drag events to not emit out. Here is a bit of the code to help explain:
```javascript
  Vue.component('string-list', {
    template: `
    <div>
      <SlickList v-model="items">
        <SlickItem v-for="(item, index) in items" :key="index" :index="index">
          <string-input v-model="item.text"></string-input>{{ item.text }}
        </SlickItem>
      </SlickList>
      <button v-on:click="createNewItem">Add New Item</button>
    </div>`,
    components: {
      SlickItem,
      SlickList
    },
    data() {
      return {
        items: this.$root.stringList
      };
    },
    methods: {
      createNewItem() {
        this.items.push({text: ''});
      }
    }
  });
```

**Comment by mattbishop** [comment]
The Vue chrome component shows that the dragged SlickItem now has an `active: null` property attached to the injected `manager` property. Before drag starts, that active property is not there.

**Comment by Jexordexan** [comment]
@mattbishop, it seems to be related to using the `:key="index"`, I changed the pen to `:key="item.text"` and it works (i wouldn't suggest that as a solution, becasue two items could have the same text). I'm not sure why that matters but I'm sure Vue is getting confused by all the indices changing. maybe you need to use an `id` on the items instead

**Comment by mattbishop** [comment]
@Jexordexan Thanks, yes you are correct. With lists of components, the key is required and must be some kind of unique constant for the item. Here is a relevant section in the docs: https://vuejs.org/v2/guide/list.html#v-for-with-a-Component

**Comment by mattbishop** [comment]
I updated my codepen with an id generator for the element list and it now all works as expected. Should this stay open to help others?

**Comment by Jexordexan** [comment]
Closing for now

**Comment by cmcleese** [comment]
> @Jexordexan Thanks, yes you are correct. With lists of components, the key is required and must be some kind of unique constant for the item. Here is a relevant section in the docs: https://vuejs.org/v2/guide/list.html#v-for-with-a-Component

I just came here to say this was very useful.
It appears that when you are managing your own custom list of components via `:key` , it must be more unique that using the `:index` that you may normally use when sorting.
Probably this is because the indexes are changing via the Mixin and its updating strangely which is potentially causing other bugs too.

Perhaps this should be mentioned in the docs.

Basically:

**BAD**
```
<SortableItem v-for="(myItem, index) in myList"
   :index="index"
   :key="index"
/>
```

**GOOD**
```
<SortableItem v-for="(myItem, index) in myList"
   :index="index"
   :key="myItem.id"
/>
```
